devito/data/data.py

```diff
@@ -155,12 +155,24 @@ def _global(self, glb_idx, decomposition):
         ret._is_distributed = any(i is not None for i in decomposition)
         return ret
 
+    def _prune_shape(self, shape):
+        # Reduce distributed MPI `Data`'s shape to that of an equivalently
+        # sliced numpy array.
+        decomposition = tuple([d for d in self._decomposition if d.size > 1])
+        retval = self.reshape(shape)
+        retval._decomposition = decomposition
```

Comment:
`reshape` creates a new `Data` I presume, so why do we have to reset the `_decomposition` here? I would expect this to happen in the constructor or at least `__array_finalize__`, so why it's not the case?

Reply:
Doesn't numpy has a no copy reshape with `view`?

Reply:
If able, the method returns a `view`:

> This will be a new view object if possible; otherwise, it will be a copy. Note there is no guarantee of the memory layout (C- or Fortran- contiguous) of the returned array.